### DISCUSSION TITLE
Programmatically change background image StreamDeck+

### DISCUSSION THREAD

**Discussion by Arr-n-D** [elgatosf/elgato#116] [Help] [answered]
Hey guys, I'm currently generating images on the fly and sending them to my StreamDeck+ plugin via websockets. I'd like to be able to change the background for the entire touch screen programmatically, is that possible? I've looked over but it seems to only be possible for actions themselves and not the entire touch screen. 

Manually changing via the right click on the touchscreen would not be a suitable solution
![image](https://github.com/elgatosf/elgato/assets/5721042/c16ad694-137a-4a17-89d4-2c12cc148aaf)

**Comment by ChekTek** [comment]
Hi @Arr-n-D, 

This is technically possible by caching the `id` of the action provided in the `willAppear` event and then referencing them later to call `setFeedback` (you can remove the id from your list in the `willDisappear` event.

You can also then bundle a profile in your plugin with actions spanning all of the touch display and then have an action that switches to this profile (and in turn triggering the `willAppear` event across all dial actions. 

https://docs.elgato.com/sdk/plugins/manifest#profiles

All of that being said, we are actively working on something we are calling "Full Canvas Control" which will enable controlling the entire Stream Deck canvas (including the touch display). This should be much easier to implement than the current profile solution. 

It's also worth noting that this is to control the image on the touch display for each action and the background image itself is not accessible via API.

**Reply by Arr-n-D** [reply, marked as answer]
Hi, I've successfully achieved what I wanted by simply modifying the profile manifest dynamically and force restarting StreamDeck, the code can be found here and suits my needs, although it's a really hacky solution and would have been better with a proper API call
https://github.com/Arr-n-D/StreamDeck-SQS-BackgroundChange
